karton/mwdb_reporter/mwdb_reporter.py

```diff
@@ -200,92 +198,103 @@ def process_config(self, mwdb: MWDB) -> MWDBConfig:
         family = (
-            self.current_task.headers["family"]
+            task.headers["family"]
             or config_data.get("family")
             or config_data.get("type", "unknown")
         )
 
-        if self.current_task.has_payload("sample"):
-            sample = self._upload_file(mwdb, self.current_task.get_payload("sample"))
-            self.log.info("[sample %s] Adding tag ripped:%s", sample.id, family)
-            sample.add_tag("ripped:" + family)
+        if task.has_payload("sample"):
+            sample = self._upload_file(task, mwdb, task.get_payload("sample"))
+            if sample:
+                self.log.info("[sample %s] Adding tag ripped:%s", sample.id, family)
+                sample.add_tag("ripped:" + family)
+            else:
+                self.log.warning("Couldn't upload sample")
         else:
             sample = None
 
-        if self.current_task.has_payload("parent"):
+        if task.has_payload("parent"):
             parent = self._upload_file(
-                mwdb, self.current_task.get_payload("parent"), parent=sample
+                task, mwdb, task.get_payload("parent"), parent=sample
             )
-            self.log.info("[sample %s] Adding tag %s", parent.id, family)
-            parent.add_tag(family)
+            if parent:
+                self.log.info("[sample %s] Adding tag %s", parent.id, family)
+                parent.add_tag(family)
+            else:
+                self.log.warning("Couldn't upload parent sample")
         else:
             parent = None
 
-        config = self._upload_config(mwdb, family, config_data, parent=parent)
+        config = self._upload_config(task, mwdb, family, config_data, parent=parent)
         return config
 
-    def process_sample(self, mwdb: MWDB) -> MWDBFile:
+    def process_sample(self, task: Task, mwdb: MWDB) -> Optional[MWDBFile]:
         """
         Processing of Sample task
 
         :param mwdb: MWDB instance
         :return: MWDBFile object or None
         """
-        if self.current_task.has_payload("parent"):
-            parent = self._upload_file(mwdb, self.current_task.get_payload("parent"))
+        if task.has_payload("parent"):
+            parent = self._upload_file(task, mwdb, task.get_payload("parent"))
         else:
             parent = None
 
-        if self.current_task.has_payload("sample"):
+        if task.has_payload("sample"):
             sample = self._upload_file(
-                mwdb, self.current_task.get_payload("sample"), parent=parent
+                task, mwdb, task.get_payload("sample"), parent=parent
             )
         else:
             sample = None
 
         return sample
 
-    def process(self) -> None:
+    def process(self, task: Task) -> None:  # type: ignore
         mwdb = self.mwdb()
-        object_type = self.current_task.headers["type"]
+        object_type = task.headers["type"]
+        mwdb_object: Optional[MWDBObject] = None
```

Comment:
This won't be enough for typing? Otherwise PyCharm marks that assignment as unreferenced
```suggestion
        mwdb_object: Optional[MWDBObject]
```